pegasus/sites.v3/code.org/public/private/edit-csp-properties/splat.haml

```diff
@@ -0,0 +1,124 @@
+-dont_cache
+
+-encrypted_app_id = request.splat_path_info[1..-1]
+-app_id = storage_decrypt_app_id(encrypted_app_id).last.to_s
+-api_base_url = "/v3/apps/#{encrypted_app_id}/shared-properties"
+
+-if rack_env == :development
+  %script{src: "/shared/js/angular/angular.js"}
+  %script{src: "/shared/js/angular-route/angular-route.js"}
+  %script{src: "/shared/js/angular-resource/angular-resource.js"}
+-else
+  %script{src: "/shared/js/angular/angular.min.js"}
+  %script{src: "/shared/js/angular-route/angular-route.min.js"}
+  %script{src: "/shared/js/angular-resource/angular-resource.min.js"}
+
+%div{"ng-app" => "propertiesApp", "ng-view" => ""}
+
+:javascript
+  'use strict';
+
+  // Declare app level module which depends on filters, and services
+  angular.module('propertiesApp', [
+    'ngRoute',
+    'ngResource',
+    'propertiesApp.controllers',
+    'propertiesApp.services'
+  ]).config(['$routeProvider', function($routeProvider) {
+    $routeProvider.when('/',
+                        {templateUrl: '/private/edit-csp-properties/properties',
+                         controller: 'PropertiesController'});
+    $routeProvider.otherwise({redirectTo: '/'});
+  }]);
+
+  // SERVICES
+  var services = angular.module('propertiesApp.services', []).
+      value('version', '0.1');
+
+  services.factory('propertiesService', ['$resource',
+    function($resource) {
+      var Property = $resource("#{api_base_url}/:key", {}, {
+        // default methods: see https://code.angularjs.org/1.2.21/docs/api/ngResource/service/$resource
+        //  'get':    {method: 'GET'},
+        //  'save':   {method: 'POST'},
+        //  'query':  {method: 'GET', isArray:true},
+        //  'remove': {method: 'DELETE'},
+        //  'delete': {method: 'DELETE'} // don't use this because it doesn't work in IE9
+      });
+
+      return Property;
+    }]);
+
+  // CONTROLLERS
+
+  var controllers = angular.module('propertiesApp.controllers', []).
+      value('version', '0.1');
+
+  controllers.controller('PropertiesController', ['$scope', '$http', '$route', '$routeParams', '$location', '$window', 'propertiesService',
+      function($scope, $http, $route, $routeParams, $location, $window, propertiesService) {
+    $scope.properties = propertiesService.get();
+    $scope.keys = null;
+    $scope.editing = [];
+    $scope.editingNewProperty = false;
+    $scope.newKey = '';
+    $scope.newValue = '';
+
+    $scope.properties.$promise.then(function(properties) {
+      console.log(properties);
```

Comment:
we might not be able to use console.log in pegasus because of ie crashing when the console is not open etc

I always forget where we have that polyfilled and where we don't